### PR TITLE
[Improve](schemaChange)schema change type adapts to other connectors

### PR DESCRIPTION
**Pull request by DongLiang-0**
# Proposed changes

When other connector types were converted to doris type before, mysql was hard-coded to be supported. Now an optimization has been made.

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java [hunk 1]
```diff
@@ -453,13 +464,31 @@ public void fillOriginSchema(JsonNode columns) {
...
+                break;
+            case ORACLE:
+                dorisTypeName = OracleType.toDorisType(sourceTypeName, length, scale);
+                break;
+            case POSTGRES:
+                dorisTypeName = PostgresType.toDorisType(sourceTypeName, length, scale);
+                break;
+            case SQLSERVER:
+                dorisTypeName = SqlServerType.toDorisType(sourceTypeName, length, scale);
+                break;
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java:483]: What about adding a unit test?

Comment (DongLiang-0) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java:490]: Of course!

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java [hunk 2]
```diff
@@ -224,44 +235,49 @@ public List<String> extractDDLList(JsonNode record) throws JsonProcessingExcepti
...
         Map<String, FieldSchema> updateFiledSchema = new LinkedHashMap<>();
         for (JsonNode column : columns) {
             buildFieldSchema(updateFiledSchema, column);
         }
         SchemaChangeHelper.compareSchema(updateFiledSchema, originFieldSchemaMap);
+        // In order to avoid operations such as rename or change, which may lead to the accidental deletion of the doris column.
+        Matcher matcher = addDropDDLPattern.matcher(ddl);
+        if (!matcher.find()) {
+            return null;
+        }
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java:261]: It is also recommended to add unit tests to this return branch, such as rename, create, truncate, etc.

Comment (DongLiang-0) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java:261]: Thanks for your suggestion, I have added it.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**